Bottom: 554b0e528deaadf831bb127ef1d4e7331177e812
Top:    7374e1a63877d1d7366c1d652245d608b5c6f253
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2008-10-22 00:27:56 +0100

Refresh of fix-preview-drawing-broken-by-

---

diff --git a/src/hid/gtk/gui-pinout-preview.c b/src/hid/gtk/gui-pinout-preview.c
index a4da4df..212d37a 100644
--- a/src/hid/gtk/gui-pinout-preview.c
+++ b/src/hid/gtk/gui-pinout-preview.c
@@ -153,6 +153,7 @@ pinout_set_data (GhidPinoutPreview * pinout, ElementType * element)
 static gboolean
 ghid_pinout_preview_expose (GtkWidget * widget, GdkEventExpose * event)
 {
+  extern HID ghid_hid;
   GhidPinoutPreview *pinout = GHID_PINOUT_PREVIEW (widget);
   GdkDrawable *save_drawable;
   double save_zoom;
@@ -194,7 +195,7 @@ ghid_pinout_preview_expose (GtkWidget * widget, GdkEventExpose * event)
   gdk_draw_rectangle (widget->window, gport->bg_gc, TRUE, 0, 0, da_w, da_h);
 
   /* call the drawing routine */
-  DrawElement (&pinout->element, 0);
+  hid_expose_callback (&ghid_hid, NULL, &pinout->element);
 
   gport->drawable = save_drawable;
   gport->zoom = save_zoom;
